src/common/cmdparse.h

```diff
@@ -64,11 +64,22 @@ cmd_getval(CephContext *cct, const cmdmap_t& cmdmap, const std::string& k, T& va
 // with default
 
 template <typename T>
-void
-cmd_getval(CephContext *cct, const cmdmap_t& cmdmap, const std::string& k, T& val, const T& defval)
+bool
+cmd_getval(CephContext *cct, const cmdmap_t& cmdmap, const std::string& k,
+	   T& val, const T& defval)
 {
-  if (!cmd_getval(cct, cmdmap, k, val))
+  if (cmdmap.count(k)) {
+    try {
+      val = boost::get<T>(cmdmap.find(k)->second);
+      return true;
+    } catch (boost::bad_get&) {
+      handle_bad_get(cct, k, typeid(T).name());
```

Comment:
this problem also exists in other commands handled in ``OSDMonitor::prepare_command_impl()`.

another way to fix this issue is to rethrow a runtime error with the cmd argument's keyname, and catch this exception in `OSDMonitor::prepare_command_impl()`. so we are able to take care of all commands in a single place once and for all.